napari/components/viewer_model.py

```diff
@@ -425,9 +425,12 @@ def _on_layers_change(self):
             self.dims.reset()
         else:
             ranges = self.layers._ranges
-            ndim = len(ranges)
-            self.dims.ndim = ndim
-            self.dims.set_range(range(ndim), ranges)
+            self.dims.update(
```

Comment:
True, this only fires the global event... It's a thing that should be fixed on the evented model actually. I'll see if I can do that quickly, otherwise it's better to do individual calls.

Reply:
I fixed it here and also in `Dims.reset()` and added a comment about it.